src/Microsoft.DotNet.Tools.Compiler.Native/Program.cs

```diff
@@ -10,13 +12,76 @@ public class Program
         public static int Main(string[] args)
         {
             DebugHelper.HandleDebugSwitch(ref args);
-            
-            var app = SetupApp();
-            
-            return ExecuteApp(app, args);
+
+            return ExecuteApp(args);
         }
-        
-        private static int ExecuteApp(CommandLineApplication app, string[] args)
+
+        private static ArgValues GetArgs(string[] args)
+        {
+            string inputAssembly = null;
+            string outputDirectory = null;
+            string temporaryOutputDirectory = null;
+            string configuration = null;
+            string mode = null;
+            string ilcArgs = null;
+            string ilcPath = null;
+            string appDepSdk = null;
+            string logPath = null;
+
+            IReadOnlyList<string> references = Array.Empty<string>();
+            IReadOnlyList<string> linklib = Array.Empty<string>();
+
+            try
+            {                
+                ArgumentSyntax.Parse(args, syntax =>
+                {                                        
+                    syntax.DefineOption("output", ref outputDirectory, "Output Directory for native executable.");
+                    syntax.DefineOption("temp-output", ref temporaryOutputDirectory, "Directory for intermediate files.");
+
+                    syntax.DefineOption("configuration", ref configuration, "debug/release build configuration. Defaults to debug.");
+                    syntax.DefineOption("mode", ref mode, "Code Generation mode. Defaults to ryujit.");
+
+                    syntax.DefineOptionList("reference", ref references, "Use to specify Managed DLL references of the app.");
+
+                    // Custom Extensibility Points to support CoreRT workflow TODO better descriptions
+                    syntax.DefineOption("ilcargs", ref ilcArgs, "Use to specify custom arguments for the IL Compiler.");
+                    syntax.DefineOption("ilcpath", ref ilcPath, "Use to plug in a custom built ilc.exe");
+                    syntax.DefineOptionList("linklib", ref linklib, "Use to link in additional static libs");
```

Comment:
Is this option being exposed at "dotnet compile" level? If it is, it shouldnt be. I do not think this is required and can be removed since all this is used to specify is msvcrt(d).lib to pass to linker on Windows, something that @brthor was going to do based upon the configuration (Debug/Release).

CC @schellap